File: types/balance.ts
export type MemberBalance = {
  user_id: number;
  name: string;
  meals_eaten: number;
  total_deposits: number;
  cost_share: number;
  balance: number;
};

export type BalanceResponse = {
  month: string;
  meal_rate: number;
  members: MemberBalance[];
};

File: backend/app/schemas/balance.py
from pydantic import BaseModel


class MemberBalance(BaseModel):
    user_id: int
    name: str
    meals_eaten: int
    total_deposits: float
    cost_share: float
    balance: float


class BalanceResponse(BaseModel):
    month: str
    meal_rate: float
    members: list[MemberBalance]
